packages/fetchai/skills/tac_negotiation/behaviours.py

```diff
@@ -106,7 +107,10 @@ def _register_agent(self) -> None:
         oef_search_msg.counterparty = self.context.search_service_address
         oef_search_dialogue = oef_search_dialogues.update(oef_search_msg)
         assert oef_search_dialogue is not None, "OefSearchDialogue not created."
-        self.context.outbox.put_message(message=oef_search_msg)
+        envelope_context = EnvelopeContext(skill_id=self.context.skill_id)
```

Comment:
By setting a skill_id in envelope context this skill is the only one receiving the response; without this other skills (e.g. `tac_participation`) would also receive the same envelope.